test/tests/pbs_Rrecord_resources_used.py

```diff
@@ -0,0 +1,616 @@
+        self.server.expect(JOB, {ATTR_state: 'R'}, jid3s1)
+
+        """
+        Confirm that the 'R' record is generated and the run_count is 2.
+        """
+        m = self.server.accounting_match(
+        msg='.*R;' + jid1 + '.*resources_used.*run_count=2', id=jid1, regexp=True)
+        self.assertNotEqual(m, None)
+        m = self.server.accounting_match(
+        msg='.*R;' + jid2 + '.*resources_used.*run_count=2', id=jid2, regexp=True)
+        self.assertNotEqual(m, None)
+
+
+    def test_t5(self):
+        """
+        Scenario: Job is rerun multiple times.
+        Expected outcome: Server should record last known resource usage every time the job is rerun.
+        """
+        dflt_q = self.server.default_queue
+        """
+        As user submit three jobs.
+        """
+        test = []
+        test += ['#PBS -N RequeueTest\n']
+        test += ['#PBS -l ncpus=1\n']
+        test += ['echo Starting test at `date`\n']
+        test += ['sleep 1000\n']
+
+        j1 = Job(TEST_USER)
+        j1.create_script(body=test)
+        j1.set_attributes({ATTR_r: 'y',ATTR_l + '.ncpus': 2})
+        jid1 = self.server.submit(j1)
+
+        j2 = Job(TEST_USER)
+        j2.create_script(body=test)
+        j2.set_attributes({ATTR_r: 'n',ATTR_l + '.ncpus': 2})
+        jid2 = self.server.submit(j2)
+
+        j3 = Job(TEST_USER)
+        j3.create_script(body=test)
+        j3.set_attributes({ATTR_r: 'y', ATTR_J: '1-4:1', ATTR_k: 'oe'})
+        jid3 = self.server.submit(j3)
+
+        subjobs = self.server.status(JOB, id=jid3, extend='t')
+        jid3s1 = subjobs[1]['id']
+        """
+        Verify that the jobs have started running.
+        """
+        self.server.expect(JOB, {ATTR_state: 'R', 'run_count': 1}, jid1, offset=5)
+        self.server.expect(JOB, {ATTR_state: 'R', 'run_count': 1}, jid2)
+        self.server.expect(JOB, {ATTR_state: 'B'}, jid3)
+        self.server.expect(JOB, {ATTR_state: 'R', 'run_count': 1}, jid3s1)
+
+        """
+        Verify that the accounting logs have Resource_List.<resource> but no R records.
+        """
+        m = self.server.accounting_match(msg='.*Resource_List.*', id=jid1, regexp=True)
+        self.assertNotEqual(m, None)
+        m = self.server.accounting_match(msg='.*R;' + jid1 + '.*resources_used.*', id=jid1, regexp=True)
+        self.assertIsNone(m)
+        m = self.server.accounting_match(msg='.*Resource_List.*', id=jid2, regexp=True)
+        self.assertNotEqual(m,None)
+        m = self.server.accounting_match(msg='.*R;' + jid2 + '.*resources_used.*', id=jid2, regexp=True)
+        self.assertIsNone(m)
+        m = self.server.accounting_match(msg='.*Resource_List.*', id=jid3s1, regexp=True)
+        self.assertNotEqual(m, None)
+
+        m = self.server.accounting_match(msg='.*R;' + re.escape(jid3s1) + '.*resources_used.*',
+        id=jid3s1, regexp=True)
+        self.assertIsNone(m)
+        #sleeping for 30 seconds
```

Comment:
That sleep is to allow the job to accumulate some resources. I need to update the comment.